sensing/pointcloud_preprocessor/test/test_distortion_corrector_node.cpp

```diff
@@ -173,48 +176,33 @@ class DistortionCorrectorTest : public ::testing::Test
       std::vector<double> timestamps =
         generatePointTimestamps(is_generate_points, stamp, number_of_points);
 
-      std::vector<uint8_t> data(number_of_points * pointcloud_msg.point_step);
+      sensor_msgs::PointCloud2Modifier modifier(pointcloud_msg);
+      modifier.setPointCloud2Fields(
+        4, "x", 1, sensor_msgs::msg::PointField::FLOAT32, "y", 1,
+        sensor_msgs::msg::PointField::FLOAT32, "z", 1, sensor_msgs::msg::PointField::FLOAT32,
+        "time_stamp", 1, sensor_msgs::msg::PointField::FLOAT64);
+      modifier.resize(number_of_points);
+
+      sensor_msgs::PointCloud2Iterator<float> iter_x(pointcloud_msg, "x");
+      sensor_msgs::PointCloud2Iterator<float> iter_y(pointcloud_msg, "y");
+      sensor_msgs::PointCloud2Iterator<float> iter_z(pointcloud_msg, "z");
+      sensor_msgs::PointCloud2Iterator<double> iter_t(pointcloud_msg, "time_stamp");
 
       for (size_t i = 0; i < number_of_points; ++i) {
-        std::memcpy(data.data() + i * pointcloud_msg.point_step, &points[i * 3], 3 * sizeof(float));
-        std::memcpy(
-          data.data() + i * pointcloud_msg.point_step + 12, &timestamps[i], sizeof(double));
+        *iter_x = points[i * 3];
+        *iter_y = points[i * 3 + 1];
```

Comment:
The variable `points` is not needed anymore

Reply:
Not sure what do you mean here, `points` is hardcoded value

Reply:
Thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/7833/commits/b0f46b71f8fa06f0d8e2bb4fb9d8578ef742ab26 and https://github.com/autowarefoundation/autoware.universe/pull/7833/commits/459a10ebfd7b1ab8ee71b50f693c9d61b74f3bfd